app/concepts/matestack/ui/core/app/app.js

```diff
@@ -10,12 +9,20 @@ const componentDef = {
   },
   computed: Vuex.mapState({
     asyncTemplate: state => state.pageTemplate,
+    currentPathName: state => state.currentPathName,
+    currentSearch: state => state.currentSearch,
+    currentOrigin: state => state.currentOrigin,
   }),
   mounted: function(){
+    const self = this;
     window.onpopstate = (event) => {
-      if (isNavigatingToAnotherPage(document.location, event)) {
-        this.$store.dispatch("navigateTo", {url: document.location.pathname, backwards: true} );
-      };
+      let needToNavigate = self.currentPathName !== document.location.pathname ||
```

Comment:
the intent behind extracting it to a separate file was to encapsulate the logic, make it independently testable in JS unit tests and not muddy the app.js with these implementation details. I'd still pefer that personally, otherwise I'd vote for a private function to take care of this as I think the implementation detail here makes it harder to read this function/see what it's actually doing. But that's arguably style (aka my style is small intention revealing functions that point you to the detail)

Reply:
changed that to your approach ;)